### DISCUSSION TITLE
OAuth flow

### DISCUSSION THREAD

**Discussion by haexhub** [ConduitPlatform/Conduit#856] [General]
Hi conduit team.

I've got a question regarding your OAuth flow.
At the moment it looks like this:

1. you start the auth flow by making a get request to your conduit backend ```http://localhost:3000/authentication/init/github```
2. this request creates a login url to github, with your fixed defined redirect url ```redirect_uri=http://localhost:3000/hook/authentication/github``` inside and not with my configured redirect url. (I'm aware that my configured redirect url will be triggered after this)

That's not what I would expect, because that means, I have to make this route public to be accessible from the internet, which I don't want. I would like to keep conduit isolated in the local network behind a proxy, API gateway or loadbalancer or all of them. 

I would like to be able, to set my own redirect url, get the code and state from the response by myself and do the get request to ```hook/authentication/github``` by my own. Is that somehow possible?

**Comment by kkopanidis** [comment]
Hi @haexhub, the base url is set via the "router url" setting of the admin panel. You can add there your api gateway as long as it is forwarded back to the conduit router. On v0.16 which is currently at the rc stage, there is also the added option to override the redirect on request, but this is not available in 0.15 versions.

When we use it in production, we have similar requirements as you specify, so we set the router url to our gateway as I write above.
